src/util/string2int.cpp

```diff
@@ -89,3 +91,77 @@ unsigned long long int unsafe_string2unsignedlonglong(
 {
   return str2number<unsigned long long int>(str.c_str(), base, false);
 }
+
+template <typename do_conversiont>
+auto wrap_string_conversion(do_conversiont do_conversion)
+  -> optionalt<decltype(do_conversion())>
+{
+  try
+  {
+    return do_conversion();
+  }
+  catch(const std::invalid_argument &)
+  {
+    return nullopt;
+  }
+  catch(const std::out_of_range &)
+  {
+    return nullopt;
+  }
+}
+
+optionalt<int> string2optional_int(const std::string &str, int base)
```

Comment:
These should be templates, as is done for `numeric_cast`.

Reply:
Ok, I'll write one